### PR TITLE
read anon user role from config, remove reference to public role

### PR DESCRIPTION
**Pull request by willgroves**
The get_user_roles should get the role specified for the anon user in the config.py at AUTH_ROLE_PUBLIC. Without this fix, the list of tables the anon user can see is limited to the tables visible to the "Public" role.

### CODE REVIEW

File: caravel/views.py [hunk 1]
```diff
@@ -176,7 +176,10 @@ def check_ownership(obj, raise_if_false=True):
 
 def get_user_roles():
     if g.user.is_anonymous():
-        return [appbuilder.sm.find_role('Public')]
+        if config.get("AUTH_ROLE_PUBLIC",None) is not None:
```
Comment (mistercrunch) [inline caravel/views.py]:
```
public_role = config.get("AUTH_ROLE_PUBLIC")
return appbuilder.sm.find_role(public_role) if public_role else []
```

**Comment by mistercrunch** [comment]
@willgroves build is broken

**Comment by xrmx** [comment]
@willgroves please rebase your changes on top of master

**Comment by willgroves** [comment]
@xrmx There is a new PR #1878 with these changes rebased onto the latest master.

**Pull request closed without merging**